sys/unix/nledl.c

```diff
@@ -57,6 +60,16 @@ nle_ctx_t *
 nle_start(const char *dlpath, nle_obs *obs, FILE *ttyrec,
           nle_seeds_init_t *seed_init)
 {
+    static bool tc_started = FALSE;
+    if (!tc_started) {
+        const char *term = getenv("TERM");
+        if (!term || tgetent(NULL, term) < 1) {
```

Comment:
Why are we sending in NULL here, instead of a buffer we can free?

Reply:
Welcome to the depth of various versions of termcap. Stallman's [doc on GNU termcap](https://www.gnu.org/software/termutils/manual/termcap-1.3/html_mono/termcap.html) says:

> If you are using the GNU version of termcap, you can alternatively ask tgetent to allocate enough space. Pass a null pointer for buffer, and tgetent itself allocates the storage using malloc. There is no way to get the address that was allocated, and you shouldn't try to free the storage.
>
> With the Unix version of termcap, you must allocate space for the description yourself and pass the address of the space as the argument buffer. There is no way you can tell how much space is needed, so the convention is to allocate a buffer 2048 characters long and assume that is enough. (Formerly the convention was to allocate 1024 characters and assume that was enough. But one day, for one kind of terminal, that was not enough.)
>
> No matter how the space to store the description has been obtained, termcap records its address internally for use when you later interrogate the description with tgetnum, tgetstr or tgetflag. If the buffer was allocated by termcap, it will be freed by termcap too if you call tgetent again. If the buffer was provided by you, you must make sure that its contents remain unchanged for as long as you still plan to interrogate the description.

(This is part of the reason termcap will always show up in memleak-checking tools like valgrind.)

However, GNU termcap isn't being used by modern distributions I believe. Instead, the manpage for [tgetent(3)](https://linux.die.net/man/3/tgetstr) says:

> The tgetent routine loads the entry for name. It returns 1 on success, 0 if there is no such entry, and -1 if the terminfo database could not be found. The emulation ignores the buffer pointer bp.

Perhaps NetHack tried to be compatible with an older / non-ncurses based version? However, they too never looked at the pointer they allocate (and it's totally unclear why that couldn't have been on the stack).

Long story short: This argument doesn't matter.